src/pages/Models/Details/ModelDetails.js

```diff
@@ -124,48 +129,61 @@ const ModelDetails = () => {
   }, [dispatch]);
 
   useEffect(() => {
-    if (modelUUID !== null && filteredModelStatusData === null) {
+    if (modelUUID !== null && modelStatusData === null) {
       // This model may not be in the first batch of models that we request
       // status from in the main loop so update the status now.
       dispatch(fetchModelStatus(modelUUID));
     }
-  }, [dispatch, modelUUID, filteredModelStatusData]);
+  }, [dispatch, modelUUID, modelStatusData]);
 
-  const handleRowClick = (e) => {
-    const currentTarget = e.currentTarget;
-    if (currentTarget.classList.contains("is-selected")) {
-      setFilterByApp("");
-      return;
-    }
-    setFilterByApp(currentTarget.dataset.app);
+  const handleAppRowClick = (e, app) => {
+    console.log(app);
+    setSlidePanelData({ app });
+    setFilterByApp(e.currentTarget.dataset.app);
```

Comment:
If the panel component is responsible for fetching its own data, as mentioned above, then none of this logic is necessary to filter the data here. It'll make this component simpler and we can get rid of more of the 'generator' logic below and have it more contained by the component.